src/System.Numerics.Vectors/tests/Matrix4x4Tests.cs

```diff
@@ -2516,5 +2516,29 @@ public unsafe void Matrix4x4FieldOffsetTest()
             Assert.Equal(new IntPtr(basePtr + 14), new IntPtr(&mat.M43));
             Assert.Equal(new IntPtr(basePtr + 15), new IntPtr(&mat.M44));
         }
+
+        [Fact]
+        public void PerspectiveFarPlaneAtInfinityTest()
+        {
+            var m = Matrix4x4.CreatePerspective(1.0f, 1.0f, 0.125f, float.PositiveInfinity);
+            Assert.False(float.IsNaN(m.M33));
```

Comment:
Should we assert that the values are `-1` and `-nearPlaneDistance` as mentioned in https://github.com/dotnet/corefx/issues/17089? Ditto for the other test cases.